File: code-coaches/medium/super-sale/super_sale.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

int get_savings(float prices[], int size)
{
    // Find the index of the most expensive item
    int max_index = 0;
    for (int i = 1; i < size; i++)
    {
        if (prices[i] > prices[max_index]) {
            max_index = i;
        }
    }

    // Remove the most expensive item by shifting the elements
    for (int i = max_index; i < size - 1; i++) {
        prices[i] = prices[i + 1];
    }

    // Calculate savings pre tax
    float discount = 0.3;
    float savings_pre_tax = 0.0;
    for (int i = 0; i < size - 1; i++) {
        savings_pre_tax += prices[i] * discount;
    }

    // Get savings post tax
    float tax = 1.07;
    float savings_post_tax = savings_pre_tax * tax;

    return floor(savings_post_tax);
}

int main()
{
    // Get input
    char input[1000];
    fgets(input, sizeof(input), stdin);

    // Split input and store each price in an array of floats
    int count = 0;
    char *token = strtok(input, ",");
    float prices[100];
    while (token != NULL)
    {
        prices[count] = atof(token);
        count++;
        token = strtok(NULL, ",");
    }

    // Print the result
    printf("%d", get_savings(prices, count));

    return 0;
}

File: code-coaches/medium/super-sale/SuperSale.swift
import Foundation

func getSavings(prices: [Float]) -> Int {
    guard var maxIndex = prices.firstIndex(of: prices.max()!) else {
        return 0
    }
    var pricesMutable = prices
    pricesMutable.remove(at: maxIndex)

    let discount: Float = 0.3
    let savingsPreTax = pricesMutable.reduce(0.0, { $0 + $1 * discount })

    let tax: Float = 1.07
    let savingsPostTax = savingsPreTax * tax

    return Int(floor(savingsPostTax))
}

func main() {
	let priceStrings = readLine()!.split(separator: ",")
	let prices = priceStrings.compactMap { Float($0) }
	let result = getSavings(prices: prices)
	print(result)
}

main()